test/text/vtt_text_parser_unit.js

```diff
@@ -586,6 +586,60 @@ describe('VttTextParser', () => {
         {periodStart: 0, segmentStart: 0, segmentEnd: 0});
   });
 
+  it('supports payload stylized', () => {
+    verifyHelper(
+        [
+          {
+            startTime: 10,
+            endTime: 20,
+            payload: 'Test',
+            fontWeight: Cue.fontWeight.BOLD,
+          },
+          {
+            startTime: 20,
+            endTime: 30,
+            payload: 'Test2',
+            fontStyle: Cue.fontStyle.ITALIC,
+          },
+          {
+            startTime: 30,
+            endTime: 40,
+            payload: 'Test3',
+            textDecoration: [Cue.textDecoration.UNDERLINE],
+          },
+          {
+            startTime: 40,
+            endTime: 50,
+            payload: 'Test4',
+          },
+          {
+            startTime: 50,
+            endTime: 60,
+            payload: 'Test5',
+            fontWeight: Cue.fontWeight.BOLD,
+          },
+          {
+            startTime: 70,
+            endTime: 80,
+            payload: 'Test<b>6</b>',
+          },
```

Comment:
These should use nestedCues to represent the internal markup.  For example:

```js
{
  startTime: 50,
  endTime: 60,
  payload: '',
  nestedCues: [
    {
      startTime: 50,
      endTime: 60,
      payload: 'Test',
      fontWeight: Cue.fontWeight.BOLD,
      fontStyle: Cue.fontStyle.NORMAL,
    },
    {
      startTime: 50,
      endTime: 60,
      payload: '5',
      fontWeight: Cue.fontWeight.BOLD,
      fontStyle: Cue.fontStyle.ITALIC,
    },
  ],
},
{
  startTime: 70,
  endTime: 80,
  payload: '',
  nestedCues: [
    {
      startTime: 70,
      endTime: 80,
      payload: 'Test',
      fontWeight: Cue.fontWeight.NORMAL,
    },
    {
      startTime: 70,
      endTime: 80,
      payload: '6',
      fontWeight: Cue.fontWeight.BOLD,
    },
  ],
},
```

The cues above that could also use `nestedCues`, but should render correctly either way.  I would imagine it would be easier to design the internal XML parsing of `VTTParser` to always emit `nestedCues`, even if there is a single markup tag on the whole payload or no markup tags at all.

Reply:
I will change to nestedCues only when necessary, because otherwise the change in the tests is very large.